src/main/java/xyz/brassgoggledcoders/transport/predicate/NamePredicate.java

```diff
@@ -23,8 +25,19 @@ public boolean test(Entity entity) {
     public static NamePredicate create(PredicateParser predicateParser) throws PredicateParserException {
         if (predicateParser.hasNextString()) {
             return new NamePredicate(predicateParser.getNextString()::equals);
+        } else if (predicateParser.hasNextRegex()) {
```

Comment:
Rather than checking directly for the Regex in the NamePredicate, I'd make the suggestion of creating the Regex as a StringPredicate (As soon below on 38) It's a separate system that allows things to act on any string passed from the system, even those created later. Right now it contains BEGINS_WITH, CONTAINS, ENDS_WITH.